ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md

```diff
@@ -0,0 +1,108 @@
+<!--
+  ~ Copyright (c) 2023-2024 Arista Networks, Inc.
+  ~ Use of this source code is governed by the Apache License 2.0
+  ~ that can be found in the LICENSE file.
+  -->
+
+# Contribution Guide for `eos_cli_config_gen` role
+
+This document outlines the steps and checklist for contributing to the `eos_cli_config_gen` role under Arista AVD.
+
+## Steps to add a new feature to eos_cli_config_gen role
+
+### Prepare development environment
+
+Follow the [Development Tooling Guide](https://avd.arista.com/5.0/docs/contribution/development-tooling.html).
+
+### Schema creation
+
+Add the schema for new feature as per EOS CLI to the appropriate schema fragments file in the `pyavd/_eos_cli_config_gen/schema/schema_fragments` directory or create a new schema file if adding a top-level feature.
+
+Please refer to the schema documentation for details on the various keys in the schema: [Schema Documentation](https://avd.arista.com/5.0/docs/contribution/input-variable-validation.html#schema-details).
+
+#### Schema Guidelines
+
+1. **Primary Key Placement:** For list-type data-models, place primary keys at the top, for readability.
+2. **Key Naming:**
+    - Follow EOS CLI for key names, when creating new schema keys.
+    - Use plural for keys that represent multiple elements (e.g., sample_policies).
+3. **Descriptions:**
+    - Only add descriptions to the keys when they provide additional context beyond the key name.
+    - Refer Arista documentation for description content.
+    - Ensure all descriptions end with punctuation.
+    - Highlight the key names in description, like - `<key_name>`.
+4. **Type Conversion:** Add `convert_types: [str]` for `type: int` keys.
```

Comment:
And convert_types: [int] for type: str if it can be numeric

Reply:
Added